src/index/IndexImpl.cpp

```diff
@@ -783,56 +789,105 @@ void IndexImpl::writeConfiguration() const {
 
 // ___________________________________________________________________________
 void IndexImpl::readConfiguration() {
-  auto f = ad_utility::makeIfstream(onDiskBase_ + CONFIGURATION_FILE);
-  f >> configurationJson_;
-  if (configurationJson_.find("git-hash") != configurationJson_.end()) {
+  ad_utility::ConfigManager config{};
+
+  // TODO Write a description.
+  std::string gitHash;
+  config.addOption("git-hash", "", &gitHash, {});
+
+  // TODO Write a description.
+  bool boolPrefixes;
+  const ad_utility::ConfigOption& prefixesOption =
+      config.addOption("prefixes", "", &boolPrefixes, false);
+
+  // TODO Write a description.
+  bool hasAllPermutations;
+  config.addOption("has-all-permutations", "", &hasAllPermutations, true);
+
+  // TODO Write a description.
+  std::vector<std::string> prefixesExternal;
+  config.addOption("prefixes-external", "", &prefixesExternal, {});
+
+  // TODO Write a description.
+  std::string lang;
+  config.addOption({"locale"s, "language"s}, "", &lang);
+
+  // TODO Write a description.
+  std::string country;
+  config.addOption({"locale"s, "country"s}, "", &country);
+
+  // TODO Write a description.
+  bool ignorePunctuation;
+  config.addOption({"locale"s, "ignore-punctuation"s}, "", &ignorePunctuation);
+
+  // TODO Write a description.
+  std::vector<std::string> languagesInternal;
+  config.addOption("languages-internal", "", &languagesInternal, {"en"});
+
+  // TODO Write a description.
+  config.addOption("num-predicates-normal", "", &numPredicatesNormal_);
+  config.addOption("num-subjects-normal", "", &numSubjectsNormal_);
+  config.addOption("num-objects-normal", "", &numObjectsNormal_);
+  config.addOption("num-triples-normal", "", &numTriplesNormal_);
+
+  // TODO Make this cleaner, than just catching all the fields of the object.
+  size_t indexFormatVersionPullRequestNumber;
+  config.addOption(
+      {"index-format-version"s, "pull-request-number"s},
+      "The number of the pull request that changed the index format most "
+      "recently.",
+      &indexFormatVersionPullRequestNumber);
+  std::string indexFormatVersionDate;
+  config.addOption({"index-format-version"s, "date"s},
+                   "The date of the last breaking change of the index format.",
+                   &indexFormatVersionDate);
+
+  // One of the options can't be done with `ConfigManager`.
```

Comment:
Which one is that?
And then you can still use `fileToJson one and then simply parse the `configurationJson_`.

Reply:
Can't remember, but I'm pretty sure, this comment is false anyway.

Reply:
Done, see [eaecacc](https://github.com/ad-freiburg/qlever/pull/1041/commits/eaecacce58ecdf0424ab77ee976011c6c23320d9) and [e87e61c](https://github.com/ad-freiburg/qlever/pull/1041/commits/e87e61c5133955df3babf81fe286dfaaa7aedf04).